content/en/tutorials/using-terraform-with-testcontainers-and-localstack/index.md

```diff
@@ -0,0 +1,372 @@
+Up until now, the supported hooks could be shell or Python scripts executed at predefined lifecycle phases — BOOT, START, READY, and SHUTDOWN. 
+By placing scripts in the respective directories (/etc/localstack/init/{stage}.d), developers can automate tasks like setting up initial states, configuring services, or performing clean-up activities.
+
+[Terraform](https://www.terraform.io/), is one of the most widely adopted tools for provisioning AWS infrastructure, so naturally, enabling Terraform configuration files to be used directly as initialization hooks boosts LocalStack's utility. 
+The direct use of Terraform scripts as init hooks allows developers to replicate production environments accurately and automate integration tests more effectively. 
+This capability ensures that the test environment mirrors the production setup as closely as possible.
+
+This tutorial guides you through using LocalStack's [**new extension**](https://github.com/localstack/localstack-extensions/tree/main/terraform-init) that supports Terraform configuration files as initialization hooks, and will show you how to leverage this new feature, and integrate it with Testcontainers for seamless testing.
+This approach simplifies the development and testing cycle, making it more efficient and closely aligned with real AWS infrastructure practices.
+
+
+## Prerequisites
+
+For this tutorial, you will need:
+
+- [LocalStack Pro](https://docs.localstack.cloud/getting-started/auth-token/) to emulate the AWS services and to use LocalStack Extensions. If you don't have LocalStack Pro yet, you can sign up on our [webapp](https://app.localstack.cloud) to get a trial license for free.
+- [Docker](https://docker.io/)
+- [LocalStack CLI](https://docs.localstack.cloud/getting-started/installation/)
+- [AWS CLI](https://aws.amazon.com/cli/)
+- Optional for building the Lambda functions: [Java 17](https://openjdk.org/install/)
+- Optional for building the Lambda functions: [Apache Maven 3.9.8](https://maven.apache.org/install.html)
+- Optional: [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) and [terraform-local](https://github.com/localstack/terraform-local)
+
+
+## Project overview
+
+You can get hands-on with this setup by cloning the [**demo repository**](https://github.com/localstack-samples/terraform-init-hooks-demo). 
+The diagram below illustrates how everything within the project connects.
+
+The application is simple, yet it reflects a realistic scenario: there's an API Gateway that directs requests to two Lambda functions.
+One Lambda function fetches product details by ID, and the other saves new products to a DynamoDB database. A CloudWatch Logs instance is used to store and access the Lambda log files.
+
+![architecture-diagram](architecture-diagram.png)
+
+## Using Terraform init hooks
```

Comment:
Can we stick with `init hooks` or `Init Hooks`?

Reply:
it's init hooks everywhere